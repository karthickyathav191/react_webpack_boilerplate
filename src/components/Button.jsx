
const Button = (props) => {
  const {
    type,
    text,
    className
  } = props
  return (
    <button className={className} type={type}>{text}</button>
  )
}

export default Button
