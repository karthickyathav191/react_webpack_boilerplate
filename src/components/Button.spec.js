import renderer from 'react-test-renderer'
import Button from './Button'
import { render, shallow } from 'enzyme'

describe("Button Test Suite", () => {
  
  it("has a valid snapshot", () => {
    const comp = renderer.create(
      <Button />
    )
    expect(comp.toJSON()).toMatchSnapshot()
  })

  it("Props pass through", () => {
    const comp = shallow(
      <Button className={'button'} text={"Click"} />
    )
    expect(comp.find(".button").text()).toEqual('Click')
  })

})