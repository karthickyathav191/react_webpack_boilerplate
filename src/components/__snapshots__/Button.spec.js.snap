// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test Suite has a valid snapshot 1`] = `<button />`;
